rustls/src/crypto/cipher/inbound.rs

```diff
@@ -106,33 +106,7 @@ impl<'a> InboundOpaque<'a> {
     }
 }
 
-/// A TLS frame, named `TLSPlaintext` in the standard.
-///
-/// This inbound type borrows its decrypted payload from the original buffer.
-/// It results from decryption.
-#[expect(clippy::exhaustive_structs)]
-#[derive(Debug)]
-pub struct InboundPlainMessage<'a> {
-    /// The content type of the message.
-    pub typ: ContentType,
-    /// The protocol version of the message.
-    pub version: ProtocolVersion,
-    /// The decrypted payload of the message.
-    pub payload: &'a [u8],
```

Comment:
The other performance reduction is due to this commit, where the replacement for `InboundPlainMessage` of `EncodedMessage<Payload>` is a little larger, but most importantly Drop. That is not necessary, really, because it is always a `Payload::Borrowed` value. That bloats out basically every function that accepts one by value.

I pushed 35a4b9d54acb6c98a30f35d67131c335e57f55a3 but it probably warrants a bit more care/squashing into this commit or another.

Reply:
Squashed and rebased, but we should both check if anything weird crept in during the rebase.